sqlglot/optimizer/scope.py

```diff
@@ -522,7 +537,9 @@ def _traverse_scope(scope):
     if isinstance(scope.expression, exp.Select):
         yield from _traverse_select(scope)
     elif isinstance(scope.expression, exp.Union):
+        yield from _traverse_ctes(scope)
```

Comment:
is this needed if we traverse_ctes inside of traverse_union?

Reply:
Yeah, had to pivot a couple of times to get this write. This handles the ctes of the top-level union scope.